### DISCUSSION TITLE
keine WLAN Verbindung zum Router nach Stromausfall

### DISCUSSION THREAD

**Discussion by t3hoe** [tbnobody/OpenDTU#1661] [Q&A]
Auf meiner OpenDTU läuft die Version 24.1.4.
Ich habe folgendes Problem, wenn nach einem Stromausfall der Router noch kein WLAN aufgebaut hat, startet die DTU im Acces-Point-Mode, soweit so gut.
Später sollte die DTU sich aber mit dem bekannten WLAN verbinden. Das passiert aber nicht. Ich habe ca. 10 Minuten gewartet.
Das AccessPoint Zeitlimit steht auf 3 Minuten.
Wenn ich mich am AP anmelde, kann ich das Anmelden am Router forcieren, in dem ich die bekannten Netzwerkeinstellungen speichere. Die DTU geht sofort in den Client-Mode.
Dann ruft sie aber auch erst die Zeit vom NTP-Server ab, wenn ich die NTP- Einstellungen speichere. Hier hatte ich aber keine Geduld länger zu warten.
Ist das ein Bug, oder mache ich etwas falsch?

**Comment by stefan123t** [comment]
@t3hoe wenn Du die WLAN Einstellungen gespeichert hast, sollte sich die OpenDTU auch mit Deinem WLAN verbinden.
Als Gegenbeweis könntest Du ja die OpenDTU vom Strom trennen / neu starten wenn der Router sein WLAN aufgebaut hat.
Wenn dann eine Verbindung zu Stande kommt, dann können wir das Problem gerne näher untersuchen.

Schau Dir dazu dann bitte die Seite https://tbnobody.github.io/OpenDTU-docs/firmware/howto/serial_console/ genauer an.
Dort findest Du Informationen wie wir die OpenDTU Boot-Logs mitschneiden können, das geht nur mit einem USB Kabel am PC/Laptop.

**Comment by t3hoe** [comment]
Neu booten bei vorhandenen WLAN ist kein Problem. Nach wenigen Sekunden, ist die OpenDTU mit WLAN verbunden, die Uhrzeit wird abgefragt und die Verbindung zum Wechselrichter hergestellt. Darum fiel es mir ja auch jetzt erst auf.
Ich werde morgen mal das Laptop anschließen und schauen, was die Console hergibt, wenn ich vor dem Booten das WLAN abschalte und danach wieder zuschalte. Danke für den Hinweis, darauf hätte ich auch selber kommen können. ;-)
Ich weiß aus leidiger Erfahrung mit den ESP's, dass das Herstellen einer WLAN-Verbindung manchmal problematisch sein kann. Ich programmiere allerdings mit der Arduino IDE in C++

**Comment by t3hoe** [comment]
Neu booten bei vorhandenen WLAN ist kein Problem. Nach wenigen Sekunden, ist die OpenDTU mit WLAN verbunden, die Uhrzeit wird abgefragt und die Verbindung zum Wechselrichter hergestellt. Darum fiel es mir ja auch jetzt erst auf.
Ich werde morgen mal das Laptop anschließen und schauen, was die Console hergibt, wenn ich vor dem Booten das WLAN abschalte und danach wieder zuschalte. Danke für den Hinweis, darauf hätte ich auch selber kommen können. ;-)
Ich weiß aus leidiger Erfahrung mit den ESP's, dass das Herstellen einer WLAN-Verbindung manchmal problematisch sein kann. Ich programmiere allerdings mit der Arduino IDE in C++

**Comment by t3hoe** [comment]
Ich habe hier einen Teil der Ausgabe heraus kopiert:
10:28:49.536 -> Starting OpenDTU
10:28:49.536 -> Initialize FS... done
10:28:49.536 -> Reading configuration... done
10:28:49.536 -> Reading PinMapping... found valid mapping done
10:28:49.536 -> Initialize Network... done
10:28:49.536 -> Setting Hostname... Configuring WiFi STA using new credentials... done
10:28:49.536 -> Initialize NTP... done
10:28:49.536 -> Initialize SunPosition... done
10:28:49.536 -> Initialize MqTT... done
10:28:49.536 -> Initialize WebApi... done
10:28:49.590 -> Initialize Display... done
10:28:49.925 -> Initialize LEDs... done
10:28:49.925 -> Check for default DTU serial... done
10:28:49.925 -> Initialize Hoymiles interface... NRF: Connection successful
10:28:49.985 ->   Setting radio PA level... 
10:28:49.985 ->   Setting DTU serial... 
10:28:49.985 ->   Setting poll interval... 
10:28:49.985 ->   Adding inverter: 11xxxxxxxx53 - Garage done
10:28:49.985 -> done
10:28:49.985 -> Switch to WiFi mode
10:28:49.985 -> Setting Hostname... done
10:28:49.985 -> Configuring WiFi STA using new credentials... done
10:28:49.985 -> Configuring WiFi STA DHCP IP... done
10:28:52.855 -> WiFi disconnected
10:28:52.855 -> Try reconnecting
10:28:52.855 -> Network lost connection
...
10:29:20.319 -> Disable search for AP... done
Die openDTU deaktiviert also recht schnell nach dem Booten die Suche nach einem Access-Point. 
Es wäre also schön, wenn wenigstens nah einiger Zeit wieder gesucht werden würde.

**Reply by tbnobody** [reply]
Es wird 30sek gesucht. Wenn nicht gefunden wird nach 10 oder 15min wieder 30 sek gesucht.

**Reply by t3hoe** [reply]
Danke für den Hinweis. Es funktioniert einwandfrei. Ich hätte einfach nur etwas warten müssen.
